python/src/dataset.rs

```diff
@@ -232,8 +232,21 @@ impl Dataset {
             .map(|idx| {
                 let dict = PyDict::new(py);
                 let schema = self_.ds.schema();
-                let field_names = schema
-                    .project_by_ids(idx.fields.as_slice())
+
+                let idx_schema = schema.project_by_ids(idx.fields.as_slice());
+
+                let is_vector = idx_schema
+                    .fields
+                    .iter()
+                    .any(|f| matches!(f.data_type(), DataType::FixedSizeList(_, _)));
```

Comment:
We don't actually record the index type in the dataset metadata. Currently, we have to try loading the index to determine if it is scalar or vector.

However, looking at the field types should be accurate today.